plugins/uefi-capsule/fu-acpi-uefi.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2023 Mario Limonciello <mario.limonciello@amd.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include <string.h>
+
+#include "fu-acpi-uefi.h"
+
+#define UEFI_HEADER_SIZE	 0x36
+#define FU_EFI_INSYDE_GUID	 "9d4bf935-a674-4710-ba02-bf0aa1758c7b"
+#define INSYDE_QUIRK_COD_WORKING 0x1
+
+struct _FuAcpiUefiTable {
+	FuFirmware parent_instance;
+	guint32 insyde_cod_status;
+	gboolean is_insyde;
+	gchar *creator_id;
+	gchar *oem_id;
+	gchar *guid;
+};
+
+struct FuAcpiInsydeQuirkSection {
+	gchar signature[6];
+	guint32 size;
+	guint32 flags;
+};
+
+G_DEFINE_TYPE(FuAcpiUefiTable, fu_acpi_uefi_table, FU_TYPE_FIRMWARE)
+
+static void
+fu_acpi_uefi_table_export(FuFirmware *firmware, FuFirmwareExportFlags flags, XbBuilderNode *bn)
+{
+	FuAcpiUefiTable *self = FU_ACPI_UEFI_TABLE(firmware);
+	fu_xmlb_builder_insert_kb(bn, "is_insyde", self->is_insyde);
+	fu_xmlb_builder_insert_kx(bn, "insyde_cod_status", self->insyde_cod_status);
+	fu_xmlb_builder_insert_kv(bn, "guid", self->guid);
+	fu_xmlb_builder_insert_kv(bn, "oem_id", self->oem_id);
+	fu_xmlb_builder_insert_kv(bn, "creator_id", self->creator_id);
+}
+
+static gboolean
+fu_acpi_uefi_parse_insyde(FuAcpiUefiTable *self, const guint8 *buf, gsize bufsz, GError **error)
+{
+	struct FuAcpiInsydeQuirkSection quirk_section;
+	guint32 data_offset = 0;
+	guint8 *data_start;
+
+	data_start = (guint8 *)g_strstr_len((gchar *)buf, bufsz, "$QUIRK");
```

Comment:
`fu_memmem_safe()` is going to be a lot quicker -- and probably more safe too :)

Reply:
I did this.